src/sentry/static/sentry/app/components/editableText.tsx

```diff
@@ -0,0 +1,192 @@
+// eslint-disable-next-line sentry/no-react-hooks
+import React, {useCallback, useEffect, useRef, useState} from 'react';
+import styled from '@emotion/styled';
+
+import TextOverflow from 'app/components/textOverflow';
+import {IconEdit} from 'app/icons/iconEdit';
+import {t} from 'app/locale';
+import space from 'app/styles/space';
+import {defined} from 'app/utils';
+import useKeypress from 'app/utils/useKeyPress';
+import useOnClickOutside from 'app/utils/useOnClickOutside';
+import Input from 'app/views/settings/components/forms/controls/input';
+import Field from 'app/views/settings/components/forms/field';
+
+type Props = {
+  text: string;
+  onSetText: (label: string) => void;
+};
+
+function EditableText({text, onSetText}: Props) {
+  const [isEditing, setIsEditing] = useState(false);
+  const [inputValue, setInputValue] = useState(text);
+
+  const isEmpty = !inputValue.trim();
+
+  const wrapperRef = useRef<HTMLDivElement>(null);
+  const contentRef = useRef<HTMLDivElement>(null);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const enter = useKeypress('Enter');
+  const esc = useKeypress('Escape');
+
+  // check to see if the user clicked outside of this component
+  useOnClickOutside(wrapperRef, () => {
+    if (isEditing && !isEmpty) {
+      onSetText(inputValue);
+      setIsEditing(false);
+    }
+  });
+
+  const onEnter = useCallback(() => {
+    if (enter && !isEmpty) {
+      onSetText(inputValue);
+      setIsEditing(false);
+    }
+  }, [enter, inputValue, onSetText]);
+
+  const onEsc = useCallback(() => {
+    if (esc) {
+      setInputValue(text);
+      setIsEditing(false);
+    }
+  }, [esc, text]);
+
+  // focus the cursor in the input field on edit start
+  useEffect(() => {
+    if (isEditing) {
+      const inputElement = inputRef.current;
+      if (defined(inputElement)) {
+        inputElement.focus();
+      }
+    }
+  }, [isEditing]);
+
+  useEffect(() => {
+    if (isEditing) {
+      // if Enter is pressed, save the text and close the editor
+      onEnter();
+      // if Escape is pressed, revert the text and close the editor
+      onEsc();
+    }
+  }, [onEnter, onEsc, isEditing]); // watch the Enter and Escape key presses
+
+  const handleInputChange = useCallback(
+    event => {
+      setInputValue(event.target.value);
+    },
+    [setInputValue]
```

Comment:
Does this prevent many re-renders? This component tree looks shallow enough that we wouldn't be preventing much.